test_optimizer.py

```
@@ -1341,3 +1399,382 @@ def _objective_diag(*, score, dd_gate_hit=False, anomaly_hit=False):
+    def test_fitness_log1p_transform_behavior(self, mock_cfg, basic_metrics, empty_rebal_log):
+        """
+        Tests log1p and copysign transformation for raw scores.
+        """
+        metrics_positive_raw = basic_metrics.copy() # Will result in positive raw
+        metrics_positive_raw["cagr"] = 100.0
+        metrics_positive_raw["max_dd"] = -5.0
+        metrics_positive_raw["turnover"] = 0.0
+        metrics_positive_raw["sortino"] = 2.5
+        
+        _, _, diag_pos = optimizer._fitness_from_metrics(metrics_positive_raw, empty_rebal_log, mock_cfg)
+        expected_raw_pos = (100.0 / ((5.0 + (0.01 * 100.0 * 2.0) + 1.0) * 1.0)) * 1.0 - 0.0 - 0.0
```

Flake8 (7.3.0)

[error] local variable 'expected_raw_pos' is assigned to but never used

(F841)